### ISSUE TITLE
onnx转paddle时报concat的错误

### ISSUE THREAD

**Issue by Eric-Zhang1990** [PaddlePaddle/X2Paddle#249]
@Channingss   我用x2paddle(0.7.3)工具将onnx (mobilenet_v2_ssd)转paddle模型时报如下错误, 根据提示信息应该是concat的错误, 但是不清楚具体原因, 请问如何解决呢? 谢谢
![WXWorkCapture_15868540459715](https://user-images.githubusercontent.com/24602893/79204776-a8ff6180-7e6f-11ea-8851-1cf7c2a9f932.png)
onnx模型百度网盘地址: 链接: https://pan.baidu.com/s/1ljmlOWyCkThK9xJYPyoZVw 提取码: s65z

**Comment by Eric-Zhang1990** [comment]
@jiangjiajun  @Channingss  你那边能否转换一下我提供的onnx模型, 是不是和我这边的错误一样呢? 谢谢

**Comment by Channingss** [comment]
@Eric-Zhang1990 好的

**Comment by Channingss** [comment]
@Eric-Zhang1990 问题是由于concat的输入中dtype存在int32和int64，在做dtype检查的时候报错。我查看了下该concat的输入，发现其中一个input(768)的前置输入是OP：shape经过一些处理，然而shape的输出是int32，shape的API：https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/shape_cn.html#paddle.fluid.layers.shape
你那边能否将其他两个输入存储为int32？
![image](https://user-images.githubusercontent.com/12471701/79407528-f8f33b00-7fcb-11ea-927a-08a715ddcf6b.png)

**Comment by Eric-Zhang1990** [comment]
@Channingss  好的, 我今天尝试一下

**Comment by Eric-Zhang1990** [comment]
@Channingss  我看了代码, 不清楚怎么将其他2个输入存储为int32呢? 代码如下:
    def forward(self, features):
        cls_logits = []
        bbox_pred = []
        for feature, cls_header, reg_header in zip(features, self.cls_headers, self.reg_headers):
            cls_logits.append(cls_header(feature).permute(0, 2, 3, 1).contiguous())
            bbox_pred.append(reg_header(feature).permute(0, 2, 3, 1).contiguous())

        batch_size = features[0].shape[0]
        cls_logits = torch.cat([c.view(c.shape[0], -1) for c in cls_logits], dim=1).view(batch_size, -1, self.cfg.MODEL.NUM_CLASSES)
        bbox_pred = torch.cat([l.view(l.shape[0], -1) for l in bbox_pred], dim=1).view(batch_size, -1, 4)

**Comment by Channingss** [comment]
@Eric-Zhang1990  self.cls_headers, self.reg_headers，cast一下？参考：https://pytorch.org/docs/stable/tensors.html#torch.Tensor.type

**Comment by jiangjiajun** [comment]
此issue长时间未更新，如有需求，可随时重新打开。
